src/codeflare_sdk/ray/cluster/cluster.py

```diff
@@ -74,10 +72,15 @@ def __init__(self, config: ClusterConfiguration):
         request.
         """
         self.config = config
-        self.app_wrapper_yaml = self.create_app_wrapper()
+        if self.config is None:
+            warnings.warn(
+                "Please provide a ClusterConfiguration to initialise the Cluster object"
+            )
```

Comment:
We can `return` early here. Then only execute the remaining code only if self.config is not None.